src/libmain.c

```diff
@@ -395,6 +395,32 @@ static void get_line_and_column_from_filename(gchar *filename, gint *line, gint
 
 
 #ifdef G_OS_WIN32
+static gint get_windows_socket_port(void)
+{
+	/* Read config file early to get TCP port number as we need it for IPC before all
+	 * other settings are read in load_settings() */
+	gchar *configfile = g_build_filename(app->configdir, "geany.conf", NULL);
+	GKeyFile *config = g_key_file_new();
+	gint port_number;
+
+	if (! g_file_test(configfile, G_FILE_TEST_IS_REGULAR))
+	{
+		geany_debug(
+			"No user config file found, use default TCP port (%s).",
+			SOCKET_WINDOWS_REMOTE_CMD_PORT);
+		g_free(configfile);
+		return SOCKET_WINDOWS_REMOTE_CMD_PORT;
+	}
+	g_key_file_load_from_file(config, configfile, G_KEY_FILE_NONE, NULL);
+	port_number = utils_get_setting_integer(config, PACKAGE, "socket_remote_cmd_port",
+		SOCKET_WINDOWS_REMOTE_CMD_PORT);
+	geany_debug("Using TCP port number %d for IPC", port_number);
+	g_free(configfile);
+	g_key_file_free(config);
```

Comment:
Would it be worth putting something like below to validate the port number?

```c
g_return_val_if_fail(port_number > 1024 && port_number <= (gint)G_MAXUINT16, SOCKET_WINDOWS_REMOTE_CMD_PORT);

Reply:
Absolutely. At some point I thought adding it and then forgot it :(.
Thanks for reminding me and even do the code :).

I lowered the minimum port value as it seems Windows has no lower boundaries for priviledged ports.
Tried port 1 and a few others < 1024 without problems.

Reply:
NP.

I don't know the convention on Win32, but ports under 1024 are ["well known" ports](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers#Well-known_ports), that's why I put that (although it was off-by-one :) Likely not necessary since Win32 doesn't seem to mind.

Reply:
Win32 might not mind, but I bet the well known service will mind :)

Reply:
Ok, we now accept port numbers from 1024 to 65535. I agree it doesn't make much sense to allow well known ports which could just cause new problems.
I added the port range also to the docs (in case anyone might read it by accident :D).